pytorch_lightning/loops/utilities.py

```diff
@@ -25,21 +25,18 @@
 from pytorch_lightning.utilities.apply_func import apply_to_collection
 from pytorch_lightning.utilities.exceptions import MisconfigurationException
 from pytorch_lightning.utilities.fetching import AbstractDataFetcher, DataLoaderIterDataFetcher
-from pytorch_lightning.utilities.finite_checks import detect_nan_parameters
 from pytorch_lightning.utilities.signature_utils import is_param_in_hook_signature
 from pytorch_lightning.utilities.types import STEP_OUTPUT
 
 
-def check_finite_loss(model: "pl.LightningModule", loss: torch.Tensor) -> None:
-    """Checks for finite parameters and loss values.
+def check_finite_loss(loss: torch.Tensor) -> None:
+    """Checks for finite loss value.
 
     Args:
-        model: a reference to the ``LightningModule``
         loss: the loss value to check to be finite
     """
-    if not torch.isfinite(loss).all():
+    if loss is not None and not torch.isfinite(loss).all():
```

Comment:
is this needed given the loss is expected to be a Tensor? when do we pass None as an input?

Reply:
I can update the type hint. 
None comes in when the training_step does not return a loss (e.g. in manual opt or when skipping a step). 

This will anyway be removed once deprecation phase of #8313 is over